nmdc_schema/identify_workflow_duplicates_emp500.py

```diff
@@ -0,0 +1,214 @@
+    {
+        "$lookup": {
+            "from": "omics_processing_set",
+            "localField": "_id",
+            "foreignField": "id",
+            "as": "omics_processing_set",
+        }
+    },
+    {"$match": {"omics_processing_set.part_of": "nmdc:sty-11-547rwq94"}},
+]
+
+
+# look for the newest assembly record, look up upstream filtering record and reads based taxonomy analysis that uses the same record
+def pick_records(omics_id):
+    pick_records_pipeline = [
+        {"$match": {"was_informed_by": omics_id}},
+        {"$sort": {"_id": -1}},
+        {"$limit": 1},
+        {
+            "$lookup": {
+                "from": "read_qc_analysis_activity_set",
+                "localField": "has_input",
+                "foreignField": "has_output",
+                "as": "upstream_filtering",
+            }
+        },
+        {
+            "$lookup": {
+                "from": "read_based_taxonomy_analysis_activity_set",
+                "localField": "has_input",
+                "foreignField": "has_input",
+                "as": "rbt_record_to_keep",
+            }
+        },
+    ]
+    records = list(asm_coll.aggregate(pick_records_pipeline))
+    return records
+
+
+# if there are multiple read based taxonomy records, only return the oldest
+
+
+def pick_rbt(input_do):
+
+    pick_oldest_rbt = [
+        {"$match": {"has_input": input_do}},
+        {"$sort": {"_id": -1}},
+        {"$limit": 1},
+    ]
+    oldest_rbt = list(rbt_coll.aggregate(pick_oldest_rbt))
+    return oldest_rbt
+
+
+# list variables for which WXA should be kept
+
+asm_keep = []
+qc_keep = []
+rbt_keep = []
+asm_delete = []
+qc_delete = []
+rbt_delete = []
+do_delete = []
+test_list = list(asm_coll.aggregate(pipeline))
+
+
+def data_objects_to_delete(output_list):
+    for do in output_list:
+        do_delete.append(do)
+
+
+def find_asm_dups(omics_identifier, asm_keep_id):
+    asm_duplicate_cursor = asm_coll.find(
+        {"was_informed_by": omics_identifier, "id": {"$not": {"$regex": asm_keep_id}}}
+    )
+    for dup_asm in asm_duplicate_cursor:
+        asm_delete.append(dup_asm["id"])
+        data_objects_to_delete(dup_asm["has_output"])
+
+
+def find_qc_dups(omics_identifier, qc_keep_id):
+    qc_duplicate_cursor = qc_coll.find(
+        {"was_informed_by": omics_identifier, "id": {"$not": {"$regex": qc_keep_id}}}
+    )
+    for qc_dup in qc_duplicate_cursor:
+        qc_delete.append(qc_dup["id"])
+        data_objects_to_delete(qc_dup["has_output"])
+
+
+def find_rbt_dups(omics_identifier, rbt_keep_id):
+    rbt_duplicate_cursor = rbt_coll.find(
+        {"was_informed_by": omics_identifier, "id": {"$not": {"$regex": rbt_keep_id}}}
+    )
+    for rbt_dup in rbt_duplicate_cursor:
+        rbt_delete.append(rbt_dup["id"])
+        data_objects_to_delete(rbt_dup["has_output"])
+
+
+for doc in test_list:
+    omics_id = doc["_id"]
+    records = pick_records(omics_id)
+    if len(list(records)) == 1:
```

Comment:
what should happen if len(list(records)) != 1?

Reply:
added else print statement